mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -171,23 +176,23 @@ object Bucketizer extends DefaultParamsReadable[Bucketizer] {
    * Binary searching in several buckets to place each data point.
    * @param splits array of split points
    * @param feature data point
-   * @param keepInvalid NaN flag.
-   *                    Set "true" to make an extra bucket for NaN values;
-   *                    Set "false" to report an error for NaN values
+   * @param keepInvalid NaN/NULL flag.
+   *                    Set "true" to make an extra bucket for NaN/NULL values;
+   *                    Set "false" to report an error for NaN/NULL values
    * @return bucket for each data point
    * @throws SparkException if a feature is < splits.head or > splits.last
    */
 
   private[feature] def binarySearchForBuckets(
       splits: Array[Double],
-      feature: Double,
+      feature: java.lang.Double,
       keepInvalid: Boolean): Double = {
-    if (feature.isNaN) {
+    if (feature == null || feature.isNaN) {
       if (keepInvalid) {
         splits.length - 1
       } else {
-        throw new SparkException("Bucketizer encountered NaN value. To handle or skip NaNs," +
-          " try setting Bucketizer.handleInvalid.")
+        throw new SparkException("Bucketizer encountered NaN/NULL values. " +
+          "To handle or skip NaNs/NULLs, try setting Bucketizer.handleInvalid.")
       }
     } else if (feature == splits.last) {
```

Comment:
could you please add some tests to validate that NULL values can now be handled in addition to NaN values by the bucketizer?